pybamm/simulation.py

```diff
@@ -967,15 +967,21 @@ def run_padding_rest(self, kwargs, rest_time, step_solution):
             step_solution,
             model,
             rest_time,
-            npts=npts,
```

Comment:
keep `npts` and raise a deprecation warning, then convert to the correct `non_linear_time`

Reply:
should I do this everywhere where `npts=npts` has been converted to `t_eval=np.linspace(0, npts)` and raise deprecation warning in `base_solver.step`?

Reply:
sounds good!